Fix logging to file and LDAP authentication

christiankreidl opened the pull request:
Hi!

This pull request fixes:

- logging to file app.log not working due to comparison of upper- and lower-case strings
- LDAP authentication not working due to problems with new config file format  
  - ConfigKeys and config file missing prefix
  - hard-coded string instead of ConfigKey used
- ActiveDirectory authentication also fixed, but could not be tested ( #856 )
- LDAP log message not an error thus changed to debug

#

christiankreidl committed:
fix: use lower-case log_level to fix case-sensitivity bug in string comparison

christiankreidl committed:
fix: add prefix to ConfigKeys key value and edit example config accordingly

christiankreidl committed:
fix: use ConfigKey instead of hard-coded name

christiankreidl committed:
fix: use ConfigKey instead of hard-coded name

christiankreidl committed:
fix: informational log message changed to more appropriate level (DEBUG instead of ERROR)

christiankreidl committed:
fix: informational log message changed to more appropriate level (DEBUG instead of ERROR)

lucs7 reviewed (commented):
@christiankreidl great PR thank you for fixing this.
I would suggest a slightly different solution for the log.php (see below).

for the issue with the Plugin Configuration requiringing he plugin name in the key, I will look into a more generalized solution, so I would suggest to remove commit 58a0ba94ef7c3b49d401961281b9c1b50c7c614b for now from the PR

lucs7 commented on lib/Common/Logging/Log.php:
```suggestion
        if ($this->log_level != 'none') {
```

lucs7 commented on lib/Common/Logging/Log.php:
```suggestion
            switch ($this->log_level) {
```

lucs7 commented on lib/Common/Logging/Log.php:
```suggestion

```

lucs7 commented on lib/Common/Logging/Log.php:
```diff
@@ -103,7 +103,7 @@ public static function Debug($message, $args = [])
     public static function Error($message, $args = [])
     {
-        $log_level = Configuration::Instance()->GetKey(ConfigKeys::LOGGING_LEVEL);
+        $log_level = strtolower(Configuration::Instance()->GetKey(ConfigKeys::LOGGING_LEVEL));
```
```suggestion

```

lucs7 committed:
fix(ldap): Rename debug configuration key for consistency

lucs7 committed:
fix(auth): Updated plugin configuration keys into nested structures
Previous changes caused an error in the plugin config loading
This commit fixes the structure and places each plugin config into its respective nested section
which was existing for some but not all plugin configs

lucs7 committed:
test(auth): Add comprehensive tests for authentication plugin configuration loading and validation

lucs7 committed:
fix(config): Preserve unknown subkeys in original structure for validation and improve error logging for invalid config values

lucs7 committed:
fix(ldap): Update default search filter to be optional with improved description

lucs7 committed:
docs(auth): Enhance description for properties in plugins
Admin Username in Active Directory configuration
LDAP Host in LDAP configuration

lucs7 committed:
docs(auth): Enhance description for Admin Username in Active Directory configuration

christiankreidl committed:
Merge pull request #1 from lucs7/fix/ldap-config

Fix/ldap config

christiankreidl committed:
Merge branch 'develop' into fix-ldap

lucs7 commented:
Looks good to me 🥳

lucs7 commented:
can be closed as #899 is taking it over
